server/autotest_server/testers/py/py_tester.py

```diff
@@ -220,6 +220,7 @@ def __init__(
         self.annotations = []
         self.overall_comments = []
         self.tags = set()
+        self.specs = specs
```

Comment:
Isn't this set in the `super().__init__` call?

Reply:
It is indeed set in the `super().__init__` call, however here it serves somewhat of a cosmetic purpose: on my machine, the pycharm linter is unable to infer the correct type of `self.schema`, so what I am doing here is trying to trick the linter.

Without setting `self.specs = specs` in the constructor of the child classes, the linter infers `self.specs` to be of type `BaseTestSpecs` which is how `self.specs` is typed in the parent class from which the testers inherit. Setting `self.specs = specs` allows the linter to infer the more specific type for the `specs` object (i.e., `PyTesterSchema`). 

This might just be pycharm, because my linter has been acting strange lately. Please let me know if you would like to remove this.

Same comment applies for all similar reviews below.

Reply:
Hi @ch-iv, yeah I don't think this is a good enough reason to add a redundant line. However, to satisfy the linter you could try adding an explicit instance attribute type annotation to each subclass instead.